src/schemas/polymesh_schema.rs

```diff
@@ -0,0 +1,140 @@
+use super::base_geom_schema::BaseGeomSchema;
+use crate::object_reader::ObjectReader;
+use crate::pod::*;
+use crate::property::*;
+use crate::reader::ArchiveReader;
+use crate::result::*;
+use crate::Archive;
+pub use std::convert::TryInto;
+
+#[derive(Debug)]
+pub struct PolyMeshSchema {
+    base_geom: BaseGeomSchema,
+    pub facecounts: ArrayPropertyReader,
+    pub faceindices: ArrayPropertyReader,
+    pub vertices: ArrayPropertyReader,
+    pub normals: Option<ArrayPropertyReader>,
+    pub uv: Option<CompoundPropertyReader>,
+    pub velocities: Option<ArrayPropertyReader>,
+}
+
+impl PolyMeshSchema {
+    pub fn new_from_object_reader(
+        object: &ObjectReader,
+        reader: &mut dyn ArchiveReader,
+        archive: &Archive,
+    ) -> Result<Self> {
+        let properties = object
+            .properties()
+            .ok_or(ParsingError::IncompatibleSchema)?;
+
+        let properties: CompoundPropertyReader = properties
+            .load_sub_property(0, reader, archive)?
+            .try_into()?;
+
+        let base_geom = BaseGeomSchema::new_from_properties(&properties, reader, archive)?;
+
+        // load required properties
+        let vertices: ArrayPropertyReader = properties
+            .load_sub_property_by_name_checked("P", reader, archive, Some(&F32X3_TYPE))?
+            .ok_or(ParsingError::IncompatibleSchema)?
+            .try_into()?;
+
+        let faceindices: ArrayPropertyReader = properties
+            .load_sub_property_by_name_checked(".faceIndices", reader, archive, Some(&I32_TYPE))?
+            .ok_or(ParsingError::IncompatibleSchema)?
+            .try_into()?;
+
+        let facecounts: ArrayPropertyReader = properties
+            .load_sub_property_by_name_checked(".faceCounts", reader, archive, Some(&I32_TYPE))?
+            .ok_or(ParsingError::IncompatibleSchema)?
+            .try_into()?;
+
+        // load optional properties
+        let normals = properties
+            .load_sub_property_by_name_checked("N", reader, archive, Some(&F32X3_TYPE))?
+            .map(|x| x.try_into())
+            .transpose()?;
+
+        let uv = properties
+            .load_sub_property_by_name("uv", reader, archive)?
+            .map(|x| -> Result<CompoundPropertyReader> { Ok(x.try_into()?) })
```

Comment:
Hint: /notSince the compiler can figure out what the resulting type is, we can actually simplify this to:
```suggestion
            .map(|x| x.try_into())
```

And we should do the same in `xform_Schema` where it was doing the same, though other files have also simplified this :)